datafusion/src/logical_plan/builder.rs

```diff
@@ -282,14 +282,42 @@ impl LogicalPlanBuilder {
             ));
         }
 
+        let left_keys = left_keys
+            .into_iter()
+            .map(|c| c.into())
+            .collect::<Vec<Column>>();
+
+        let mut swap = false;
         let left_keys: Vec<Column> = left_keys
+            .clone()
             .into_iter()
-            .map(|c| c.into().normalize(&self.plan))
-            .collect::<Result<_>>()?;
+            .map(|c| c.normalize(&self.plan))
+            .collect::<Result<_>>()
+            .or_else(|_| {
+                swap = true;
```

Comment:
Shouldn't "swapping" be done on an individual level, e.g. for `left1=right1 and right2=left2` it will swap everything now?

Reply:
Yes, thanks for you review. I will address this.

Reply:
I have refactored this to operate at an individual condition level and I think the code is much more succinct.